File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) => logLine.Substring(logLine.IndexOf(':') + 1).Trim();

    public static string LogLevel(string logLine) => logLine.Substring(1, logLine.IndexOf(']') - 1).ToLower();

    public static string Reformat(string logLine) => string.Format("{0} ({1})", Message(logLine), LogLevel(logLine));
}

File: fsharp/log-levels/LogLevels.fs
module LogLevels

open System

let toLower (x: string) = x.ToLower()
let trim (x: string) = x.Trim()
let getTokens (x: string) = x.Split([|'['; ']'; ':'|], StringSplitOptions.RemoveEmptyEntries)

let message (logLine: string): string = 
    logLine
    |> getTokens
    |> Array.tryLast
    |> function
        | Some x -> x |> trim
        | None -> ""

let logLevel(logLine: string): string =
    logLine
    |> getTokens
    |> Array.tryHead
    |> function
        | Some x -> x |> trim |> toLower
        | None -> ""

let reformat(logLine: string): string = 
    $"{message logLine} ({logLevel logLine})"
